packages/grid/x-data-grid/src/hooks/features/filter/useGridFilter.tsx

```diff
@@ -40,12 +40,44 @@ export const filterStateInitializer: GridStateInitializer<
     ...state,
     filter: {
       filterModel: sanitizeFilterModel(filterModel, props.disableMultipleColumnsFiltering, apiRef),
-      visibleRowsLookup: {},
       filteredDescendantCountLookup: {},
     },
   };
 };
 
+function useGridVisibleRowsLookup(apiRef: React.MutableRefObject<GridPrivateApiCommunity>) {
```

Comment:
I'm going to move it outside of the `useGridFilter` hook, it was just faster to do it like this to validate the approach

Reply:
Done